include/nicegraf.h

```diff
@@ -3063,7 +3133,8 @@ ngf_error ngf_submit_cmd_buffers(uint32_t nbuffers, ngf_cmd_buffer* bufs) NGF_NO
 ngf_error ngf_cmd_begin_render_pass(
     ngf_cmd_buffer              buf,
     const ngf_render_pass_info* pass_info,
-    ngf_render_encoder*         enc) NGF_NOEXCEPT;
+    ngf_render_encoder*         enc,
```

Comment:
2.
The signatures of `ngf_cmd_begin_render_pass` and `ngf_cmd_compute_pass` should not change.

Instead, let's add two new functions in `include/nicegraf-mtl-handles.h`:

`void ngf_mtl_set_sample_attachment_for_next_render_pass(ngf_cmd_buffer cmd_buf, uintptr_t sample_buf_attachment_descriptor);`

and

`void ngf_mtl_set_sample_attachment_for_next_compute_pass(ngf_cmd_buffer cmd_buf, uintptr_t sample_buf_attachment_descriptor);`

they both receive the command buffer handle as the first argument, and a pointer as their second argument.

For the first one, that second argument should be a pointer to a `MTLRenderPassSampleBufferAttachmentDescriptor` and for the second it should be a pointer to a `MTLComputePassSampleBufferAttachmentDescriptor`